osbuild/pipeline.py

```diff
@@ -333,10 +329,10 @@ def run(self, store, interactive=False, libdir=None, secrets=None, output_direct
             # missing, since it is not a mandatory part of the result and would
             # usually be needless overhead.
             if object_store.contains(self.output_id):
-                results = {"output_id": self.output_id,
-                           "success": True}
-                if object_store.contains(self.tree_id):
-                    results["tree_id"] = self.tree_id
+                results = {"success": True}
+                if output_directory:
+                    with object_store.new(base_id=self.output_id) as output:
+                        output.export(output_directory)
```

Comment:
I think I mentioned this before, but with this change it I think I can mention it again :) What if we instead of `contains` followed by `new` we have a `object_store.get` that gives you the object or `None` and then just use this. I think we should get this in as is and I am happy to have a go at this idea later today and see how that looks like.

Reply:
Sounds good! This is the only caller, I think, so I did not investigate so far. But I like the idea!